src/jdk.jpackage/share/classes/jdk/jpackage/internal/AppImageFile.java

```diff
@@ -226,10 +247,16 @@ public static Document readXml(Path appImageDir) throws IOException {
                     "warning.invalid-app-image"), appImageDir));
 
         }
+        // this should never be the case, but maintaining behavior of
+        // creating default launchers without AppImageFile present
 
-        launchers.add(APP_NAME.fetchFrom(params));
+        List<String> names = new ArrayList<>();
+        names.add(APP_NAME.fetchFrom(params));
         ADD_LAUNCHERS.fetchFrom(params).stream().map(APP_NAME::fetchFrom).forEach(
-                launchers::add);
+                names::add);
+        for (var name : names) {
+            launchers.add(new LauncherInfo(name, true, true));
+        }
```

Comment:
This block of code can be simplified down to:
`ADD_LAUNCHERS.fetchFrom(params).stream().map(APP_NAME::fetchFrom).map(name -> new LauncherInfo(name, true, true)).forEach(launchers::add);`